lib/presentation/widget/myspace/my_space_viewmodel.dart

```diff
@@ -66,7 +67,29 @@ class MySpaceViewModel extends BaseViewModel {
       this._downloadFileIOSInteractor
   ) : super(store);
 
-  void downloadFile(DocumentId documentId) {
+  ThunkAction<AppState> handleDownloadFile(DocumentId documentId) {
+    return (Store<AppState> store) async {
+      final status = await Permission.storage.status;
+      switch (status) {
+        case PermissionStatus.granted: _download(documentId);
+        break;
+        case PermissionStatus.permanentlyDenied: _appNavigation.popBack();
```

Comment:
[Confirm] Int the case`permanentlyDenied`. Should we notify user about this situation by showing toast?

Reply:
yes, I will contact to designer for more details

Reply:
I create new ticket for this: https://ci.linagora.com/linagora/lgs/linshare/products/linshare-mobile-apps/issues/630. Please do it later. thanks